client/components/language-picker/test/modal.js

```diff
@@ -451,4 +452,83 @@ describe( 'LanguagePickerModal', () => {
 			LanguagePickerModal.prototype.getLanguagesListColumnsCount = getLanguagesListColumnsCount;
 		} );
 	} );
+
+	describe( 'search event', () => {
+		let recordTracksEvent;
+		let wrapper;
+
+		beforeEach( () => {
+			recordTracksEvent = jest.fn();
+
+			wrapper = shallow(
+				<LanguagePickerModal { ...defaultProps } recordTracksEvent={ recordTracksEvent } />
+			);
+		} );
+
+		test( 'should not fire when language does not change', () => {
+			expect( wrapper.state( 'selectedLanguageSlug' ) ).toEqual( defaultProps.selected );
+
+			wrapper.instance().handleClose();
+
+			expect( recordTracksEvent ).not.toHaveBeenCalled();
```

Comment:
I'd like to note that these kinds of tests will work until we add a second event to track. Perhaps to tighten them we could check the arguments we've NOT called `recordTracksEvent` with?